extras/exporters/gltf-exporter.js

```diff
@@ -348,33 +375,32 @@ class GltfExporter extends CoreExporter {
                 // all mesh instances of a single node are stores as a single gltf mesh with multiple primitives
                 const meshInstances = entityMeshInstances.meshInstances;
                 meshInstances.forEach((meshInstance) => {
-                    const indexBuffer = meshInstance.mesh.indexBuffer[0];
-                    const vertexBuffer = meshInstance.mesh.vertexBuffer;
-                    const vertexFormat = vertexBuffer.getFormat();
-                    const numVertices = vertexBuffer.getNumVertices();
 
                     const primitive = {
                         attributes: {},
                         material: resources.materials.indexOf(meshInstance.material)
                     };
                     mesh.primitives.push(primitive);
 
-                    // An accessor is a vertex attribute
-                    const writeAccessor = (element) => {
+                    // vertex buffer
+                    const vertexBuffer = meshInstance.mesh.vertexBuffer;
+                    const numVertices = vertexBuffer.getNumVertices();
+                    const vertexFormat = vertexBuffer.getFormat();
+                    const interleaved = vertexFormat.interleaved;
+                    vertexFormat.elements.forEach((element, elementIndex) => {
```

Comment:
```suggestion
                    // vertex buffer
                    const { vertexBuffer } = meshInstance.mesh;
                    const { numVertices, format } = vertexBuffer;
                    const { interleaved, elements } = format;
                    elements.forEach((element, elementIndex) => {
```

I guess this is kind of a taste thing, I don't know why you prefer `vertexBuffer.getNumVertices()` over `vertexBuffer.numVertices`, the `fake getter` looks longer than necessary to me. And using the properties allows to remove one line and every line looks shorter than before.

Reply:
numVertices is private, and as much as possible I avoid access to those from other classes.

Reply:
Thank you, that explains it, I thought it would be `_numVertices` then.

Reply:
> I thought it would be `_numVertices` then.

We consider members which are not documented using JSDocs to be private, regardless of _, and as we develop the engine, we're free to change those as needed (rename / remove ..). If something has JSDocs and is exposed in public documentation, those are public and we take care to avoid their changes.

So here I prefer to use documented functions, limiting scope for broken code in the future.

and thanks for the getSource(), change to that as well.